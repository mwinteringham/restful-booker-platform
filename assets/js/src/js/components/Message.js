import React, { useEffect, useState } from 'react';
import ReactModal from 'react-modal';
import { API } from '../libs/Api';

const Message = ({messageId, closeMessage, refreshMessageList}) => {

    const [message, setMessage] = useState({
            name : "",
            email : "",
            phone : "",
            subject : "",
            description : ""
        })

    useEffect(() => {
        API.putMessageRead(messageId);
        API.getMessage(messageId, setMessage);
    }, [])
    
    return <ReactModal 
                isOpen={true}
                ariaHideApp={false}
                contentLabel="onRequestClose Example"
                className="message-modal" >
                
                <div data-testid="message">
                    <div className="form-row">
                        <div className="col-10">
                            <p><span>From: </span>{message.name}</p>
                        </div>
                        <div className="col-2">
                            <p><span>Phone: </span>{message.phone}</p>
                        </div>
                    </div>
                    <div className="form-row">
                        <div className="col-12">
                            <p><span>Email: </span>{message.email}</p>
                        </div>
                    </div>
                    <div className="form-row">
                        <div className="col-12">
                            <p><span>{message.subject}</span></p>
                        </div>
                    </div>
                    <div className="form-row">
                        <div className="col-12">
                            <p>{message.description}</p>
                        </div>
                    </div>
                    <div className="form-row">
                        <div className="col-12">
                            <button className="btn btn-outline-primary" onClick={() => closeMessage()}>Close</button>
                        </div>
                    </div>
                </div>
            </ReactModal>

}

export default Message;
