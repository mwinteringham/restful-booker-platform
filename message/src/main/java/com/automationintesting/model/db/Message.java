package com.automationintesting.model.db;

import com.fasterxml.jackson.annotation.JsonProperty;

import javax.persistence.Entity;
import javax.validation.constraints.*;
import java.sql.ResultSet;
import java.sql.SQLException;

@Entity
public class Message {

    @JsonProperty
    private int messageid;

    @JsonProperty
    @NotNull(message = "Name must be set")
    @NotBlank (message = "Name may not be blank")
    private String name;

    @JsonProperty
    @Email
    @NotNull(message = "Email must be set")
    @NotBlank(message = "Email may not be blank")
    private String email;

    @JsonProperty
    @Size(min = 11, max = 21, message = "Phone must be between 11 and 21 characters.")
    @NotNull(message = "Phone must be set")
    @NotBlank(message = "Phone may not be blank")
    private String phone;

    @JsonProperty
    @Size(min = 5, max = 100, message = "Subject must be between 5 and 100 characters.")
    @NotNull(message = "Subject must be set")
    @NotBlank(message = "Subject may not be blank")
    private String subject;

    @JsonProperty
    @Size(min = 20, max = 2000, message = "Message must be between 20 and 2000 characters.")
    @NotNull(message = "Message must be set")
    @NotBlank(message = "Message may not be blank")
    private String description;

    public Message() {
    }

    public Message(String name, String email, String phone, String subject, String description) {
        this.name = name;
        this.email = email;
        this.phone = phone;
        this.subject = subject;
        this.description = description;
    }

    public Message(ResultSet result) throws SQLException {
        this.messageid = result.getInt("messageid");
        this.name = result.getString("name");
        this.email = result.getString("email");
        this.phone = result.getString("phone");
        this.subject = result.getString("subject");
        this.description = result.getString("description");
    }

    public int getMessageid() {
        return messageid;
    }

    public String getName() {
        return name;
    }

    public String getEmail() {
        return email;
    }

    public String getPhone() {
        return phone;
    }

    public String getSubject() {
        return subject;
    }

    public String getDescription() {
        return description;
    }

    public void setMessageid(int messageid) {
        this.messageid = messageid;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public void setSubject(String subject) {
        this.subject = subject;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    @Override
    public String toString() {
        return "Message{" +
                "messageid=" + messageid +
                ", name='" + name + '\'' +
                ", email='" + email + '\'' +
                ", phone='" + phone + '\'' +
                ", subject='" + subject + '\'' +
                ", description='" + description + '\'' +
                '}';
    }
}
