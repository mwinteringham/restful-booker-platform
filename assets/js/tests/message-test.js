import React from 'react';
import Message from '../src/js/components/Message';
import nock from 'nock';
import '@testing-library/jest-dom'
import {
    render,
    waitFor,
    asFragment,
    getByText,
    screen
  } from '@testing-library/react'

nock('http://localhost')
    .persist()
    .get('/message/1')
    .reply(200, {
            "name" : "Mark Winteringham",
            "email" : "mark@mwtestconsultancy.co.uk",
            "phone" : "01821 912812",
            "subject" : "Subject description here",
            "description" : "Lorem ipsum dolores est"
        }
    );

nock('http://localhost')
    .persist()
    .put('/message/1/read')
    .reply(200);

test('Message popup is populated with details', async () => {
    render(<Message messageId={1} />);

    const modalComponent = screen.getByTestId(/message/);
    expect(modalComponent).toMatchSnapshot();
});