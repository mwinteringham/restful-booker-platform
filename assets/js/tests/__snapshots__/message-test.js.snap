// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message popup is populated with details 1`] = `
<div
  data-testid="message"
>
  <div
    class="form-row"
  >
    <div
      class="col-10"
    >
      <p>
        <span>
          From: 
        </span>
      </p>
    </div>
    <div
      class="col-2"
    >
      <p>
        <span>
          Phone: 
        </span>
      </p>
    </div>
  </div>
  <div
    class="form-row"
  >
    <div
      class="col-12"
    >
      <p>
        <span>
          Email: 
        </span>
      </p>
    </div>
  </div>
  <div
    class="form-row"
  >
    <div
      class="col-12"
    >
      <p>
        <span />
      </p>
    </div>
  </div>
  <div
    class="form-row"
  >
    <div
      class="col-12"
    >
      <p />
    </div>
  </div>
  <div
    class="form-row"
  >
    <div
      class="col-12"
    >
      <button
        class="btn btn-outline-primary"
      >
        Close
      </button>
    </div>
  </div>
</div>
`;
